pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -181,28 +181,31 @@ func (ctrl *csiSnapshotSideCarController) getCSISnapshotInput(content *crdv1.Vol
 	klog.V(5).Infof("getCSISnapshotInput for content [%s]", content.Name)
 	var class *crdv1.VolumeSnapshotClass
 	var err error
+
+	// Resolve snapshotting secret credentials.
+	// We should perform this check before querying the snapshot class, so
+	// that the content can be deleted if this class no longer exists.
+	snapshotterCredentials, err := ctrl.GetCredentialsFromAnnotation(content)
+	if err != nil {
+		return nil, nil, err
```

Comment:
This will affect snapshot creation.  We need to have SnapshotClass for snapshot creation.

Instead of making changes to `getCSISnapshotInput`, we can make a change in `deleteCSISnapshotOperation`.
Replace the call `_, snapshotterCredentials, err := ctrl.getCSISnapshotInput(content)` with `snapshotterCredentials, err := ctrl.GetCredentialsFromAnnotation(content)` as we only need to get credentials from annotation at delete snapshot time.

Reply:
Thanks for the quick review! I'll make the changes now.